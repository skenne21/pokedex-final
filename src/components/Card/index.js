import React from 'react';

const Card = ({ type, getPokemonInfo, pokemons }) => {
  console.log(pokemons)
  const { name, pokemon } = type
  return (
    <div>
      <h1>{name}</h1>
      <button onClick={ () => getPokemonInfo(pokemon)}></button>
    </div>
  )

}

export default Card