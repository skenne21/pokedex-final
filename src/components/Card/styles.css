.Card {
  width: 25%;
  margin: 5px;
  display: inline-block;
}

.Card button {
  font-size: 1em;
  width: 60%;
  height: 80px;
}