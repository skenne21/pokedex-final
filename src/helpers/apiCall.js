export const getTypes =  async () => {
  try {
    const response = await fetch('http://localhost:3001/types');
    const apiData = await response.json();
    return apiData;
  } catch (error) {
    throw error
  }
}

export const getPokemon = async (ids) => {
  const pokemonPromises = ids.map(async pokemon => {
    const apiData = await fetchPokemon(pokemon)
    const indivdualPokemon = {
      name: apiData.name,
      weight: apiData.weight,
      image: apiData.sprites.back_default
    }
    return indivdualPokemon;
  })  
  return Promise.all(pokemonPromises)
}

export const fetchPokemon = async (pokemon) => {
  try {
    const response = await fetch(`http://localhost:3001/pokemon/${pokemon}`);
    const apiData = await response.json();
    return apiData;
  } catch (error) {
    throw error
  }  
}


