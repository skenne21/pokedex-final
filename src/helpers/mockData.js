export const types = [
  {
    id:"1",
    name: "normal",
    pokemon: [
      '16',
      '17'
    ]
  }, {
    id: "2",
    name: "fighting",
    pokemon: [
      "56",
      "57",
      "66",
    ]
  }
]