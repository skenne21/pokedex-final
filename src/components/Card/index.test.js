import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';
import * as mocks from '../../helpers/mockData';

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card
        type={mocks.types}
      />
    );
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})