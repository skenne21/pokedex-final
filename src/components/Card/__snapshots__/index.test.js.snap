// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    type={
      Array [
        Object {
          "id": "1",
          "name": "normal",
          "pokemon": Array [
            "16",
            "17",
          ],
        },
        Object {
          "id": "2",
          "name": "fighting",
          "pokemon": Array [
            "56",
            "57",
            "66",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
